server/cli/src/cli.rs

```diff
@@ -396,6 +405,102 @@ async fn main() -> anyhow::Result<()> {
             info!("Refresh data result: {:#?}", result);
         }
         Action::SignPlugin { path, key, cert } => sign_plugin(&path, &key, &cert)?,
+        Action::UpsertStandardReports { name } => {
+            let connection_manager = get_storage_connection_manager(&settings.database);
+            let con = connection_manager.connection()?;
+            let base_reports_dir = "./reports";
+
+            let report_names = fs::read_dir(base_reports_dir)?
+                .map(|res| res.map(|e| e.path().into_os_string().into_string().unwrap()))
+                .collect::<Result<Vec<String>, std::io::Error>>()?;
+
+            let report_names = match &name {
+                Some(name) => report_names
+                    .into_iter()
+                    .filter(|report| report == name)
+                    .collect(),
+                None => report_names,
+            };
+
+            for name_dir in report_names {
+                let report_versions = fs::read_dir(&name_dir)?
+                    .map(|res| res.map(|e| e.path().into_os_string().into_string().unwrap()))
+                    .collect::<Result<Vec<String>, std::io::Error>>()?;
+
+                let (_, name) = name_dir.rsplit_once('/').unwrap();
+
+                for version_dir in report_versions {
+                    let (_, version) = version_dir.rsplit_once('/').unwrap();
+
+                    let args = BuildArgs {
+                        dir: format!("{version_dir}/src"),
+                        output: format!("{version_dir}/generated/{name}.json").into(),
+                        template: "template.html".to_string(),
+                        header: None,
+                        footer: None,
+                        query_gql: Some("query.graphql".to_string()),
+                        query_default: None,
+                        query_sql: None,
+                    };
```

Comment:
later this will need to be more specific to include headers and footers, and also to be flexible to different types of standard report queries.
Some of these might just need renaming to a standard graphql name, others will need to search and pass a vec of sql queries. (You can see this in item-usage report which has a large number of sql queries)

Reply:
Have added an issue for this #4843

Reply:
I think the idea was to specify the header, footer, queries, etc all in the manifest file. Then you should only need to read the one file here? Or at least get the paths from the manifiest to add to pass to the buildier?